avocado/plugins/runner_nrunner.py

```diff
@@ -27,16 +32,42 @@ class Runner(RunnerInterface):
     name = 'nrunner'
     description = '*EXPERIMENTAL* nrunner based implementation of job compliant runner'
 
+    def _save_to_file(self, filename, buff, mode='wb'):
+        with open(filename, mode) as fp:
+            fp.write(buff)
+
+    def _populate_task_logdir(self, base_path, task, statuses, debug=False):
+        # We are copying here to avoid printing duplicated information
+        local_statuses = copy(statuses)
+        last = local_statuses[-1]
+        stdout = last.pop('stdout')
```

Comment:
You're counting on the last status containing an `stdout` and `stderr`, which may not be the case. Again, thinking about other kinds of tests, they may never have anything that resembles `stdout` and `stderr`. A `web-service` kind of test may be such a case.

Anyway, this results when running `./examples/job/nrunner.py`:

```
JOB ID     : a6b07e59fa2641b3ec99b4f3bcff506b79f312d3
JOB LOG    : /home/cleber/avocado/job-results/job-2020-04-25T17.47-a6b07e5/job.log
 (1/13) 1-selftests.unit.test_resolver.ReferenceResolution.test_no_args: |
Avocado crashed: KeyError: 'stdout'
Traceback (most recent call last):

  File "/home/cleber/src/avocado/avocado/avocado/core/job.py", line 616, in run_tests
    execution_order)

  File "/home/cleber/src/avocado/avocado/avocado/plugins/runner_nrunner.py", line 113, in run_suite
    job.config.get('core.debug'))

  File "/home/cleber/src/avocado/avocado/avocado/plugins/runner_nrunner.py", line 43, in _populate_task_logdir
    stdout = last.pop('stdout')

KeyError: 'stdout'

Please include the traceback info and command line used on your bug report
Report bugs visiting https://github.com/avocado-framework/avocado/issues/new
```

Reply:
Something like this seems to do the trick:

```diff
diff --git a/avocado/plugins/runner_nrunner.py b/avocado/plugins/runner_nrunner.py
index 17cd013e..344da5a0 100644
--- a/avocado/plugins/runner_nrunner.py
+++ b/avocado/plugins/runner_nrunner.py
@@ -40,18 +40,26 @@ class Runner(RunnerInterface):
         # We are copying here to avoid printing duplicated information
         local_statuses = copy(statuses)
         last = local_statuses[-1]
-        stdout = last.pop('stdout')
-        stderr = last.pop('stderr')
+        try:
+            stdout = last.pop('stdout')
+        except KeyError:
+            stdout = None
+        try:
+            stderr = last.pop('stderr')
+        except KeyError:
+            stderr = None
 
         # Create task dir
         task_path = os.path.join(base_path, task.identifier.replace('/', '_'))
         os.makedirs(task_path, exist_ok=True)
 
         # Save stdout and stderr
-        stdout_file = os.path.join(task_path, 'stdout')
-        stderr_file = os.path.join(task_path, 'stderr')
-        self._save_to_file(stdout_file, stdout)
-        self._save_to_file(stderr_file, stderr)
+        if stdout is not None:
+            stdout_file = os.path.join(task_path, 'stdout')
+            self._save_to_file(stdout_file, stdout)
+        if stderr is not None:
+            stderr_file = os.path.join(task_path, 'stderr')
+            self._save_to_file(stderr_file, stderr)
 
         # Save debug
         if debug:
```

Reply:
Right, will fix in v2. Thanks for the patch.